File: jphp-javafx-ext/src/main/java/org/develnext/jphp/ext/javafx/classes/layout/UXAnchorPane.java
package org.develnext.jphp.ext.javafx.classes.layout;

import javafx.scene.Node;
import javafx.scene.layout.AnchorPane;
import javafx.scene.layout.Pane;
import org.develnext.jphp.ext.javafx.JavaFXExtension;
import php.runtime.annotation.Reflection.Name;
import php.runtime.annotation.Reflection.Signature;
import php.runtime.env.Environment;
import php.runtime.reflection.ClassEntity;

@Name(JavaFXExtension.NAMESPACE + "layout\\UXAnchorPane")
public class UXAnchorPane extends UXPane {
    interface WrappedInterface {

    }

    public UXAnchorPane(Environment env, Pane wrappedObject) {
        super(env, wrappedObject);
    }

    public UXAnchorPane(Environment env, ClassEntity clazz) {
        super(env, clazz);
    }

    @Signature
    public void __construct() {
        __wrappedObject = new AnchorPane();
    }

    @Signature
    public static void setLeftAnchor(Node node, double value) {
        AnchorPane.setLeftAnchor(node, value);
    }

    @Signature
    public static void setRightAnchor(Node node, double value) {
        AnchorPane.setRightAnchor(node, value);
    }

    @Signature
    public static void setTopAnchor(Node node, double value) {
        AnchorPane.setTopAnchor(node, value);
    }

    @Signature
    public static void setBottomAnchor(Node node, double value) {
        AnchorPane.setBottomAnchor(node, value);
    }

    @Signature
    public static double getLeftAnchor(Node node) {
        return AnchorPane.getLeftAnchor(node);
    }

    @Signature
    public static double getRightAnchor(Node node) {
        return AnchorPane.getRightAnchor(node);
    }

    @Signature
    public static double getTopAnchor(Node node) {
        return AnchorPane.getTopAnchor(node);
    }

    @Signature
    public static double getBottomAnchor(Node node) {
        return AnchorPane.getBottomAnchor(node);
    }
}

File: exts/jphp-gui-ext/src/main/resources/JPHP-INF/sdk/php/gui/layout/UXAnchorPane.php
<?php
namespace php\gui\layout;

use php\gui\layout\UXPane;
use php\gui\UXNode;

/**
 * Class UXAnchorPane
 * @package php\gui\layout
 * @packages gui, javafx
 */
class UXAnchorPane extends UXPane
{
    /**
     * @param UXNode $node
     * @return double
     */
    static function getLeftAnchor(UXNode $node) {}

    /**
     * @param UXNode $node
     * @return double
     */
    static function getRightAnchor(UXNode $node) {}

    /**
     * @param UXNode $node
     * @return double
     */
    static function getTopAnchor(UXNode $node) {}

    /**
     * @param UXNode $node
     * @return double
     */
    static function getBottomAnchor(UXNode $node) {}

    /**
     * @param UXNode $node
     * @param double $value
     */
    static function setLeftAnchor(UXNode $node, $value) {}

    /**
     * @param UXNode $node
     * @param double $value
     */
    static function setRightAnchor(UXNode $node, $value) {}

    /**
     * @param UXNode $node
     * @param double $value
     */
    static function setTopAnchor(UXNode $node, $value) {}

    /**
     * @param UXNode $node
     * @param double $value
     */
    static function setBottomAnchor(UXNode $node, $value) {}

    /**
     * @param UXNode $node
     * @param double $value
     */
    static function setAnchor(UXNode $node, $value) {}
}